crates/bevy_ecs/src/component.rs

```diff
@@ -177,6 +182,9 @@ pub struct ComponentDescriptor {
     // this descriptor describes.
     // None if the underlying type doesn't need to be dropped
     drop: Option<for<'a> unsafe fn(OwningPtr<'a>)>,
+    // SAFETY: this function must be safe to call with pointers pointing to items of the type
+    // this descriptor describes. Must never panic.
+    swap: for<'a> unsafe fn(Ptr<'a>, Ptr<'a>),
```

Comment:
The swap function should operate on a pair of PtrMut

Reply:
Unless we can prove that both of these are unique, I don't think we can.